fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/ReindexManager.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl.impl;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+
+/**
+ * Class to coordinate the index rebuilding tasks.
+ * @author whikloj
+ */
+public class ReindexManager {
+
+    private static final Logger LOGGER = getLogger(ReindexManager.class);
+
+    private final String transactionId;
+
+    private final List<ReindexWorker> workers;
+
+    private final Iterator<String> ocflIter;
+
+    private final Map<String, String> resultStates = new HashMap<>();
```

Comment:
This seems to be less of a state map and more of a map of errors, the successes have an empty value. It also has the potential to grow very large. How are we planning on using these results? For me, what makes sense is to have a count of the objects that were successfully processed, and either a collection or a map of exceptions that occurred during processing. Or, we could just keep a count of both and immediately log errors so that we do not need to collect them. I think keeping the counts is worth while because it will allow us to do periodic reporting during the processing.